application/features/Connection.py

```
@@ -49,6 +53,14 @@
                 ("127.0.0.1", self.server.chain_port),
             )
         )
+        logger.debug(
+            "Connected!  Tunnel open %r -> %r -> %r"
+            % (
+                self.request.getpeername(),
+                chan.getpeername(),
+                ("127.0.0.1", self.server.chain_port),
+            )
```

Ruff

Use format specifiers instead of percent format

Replace with format specifiers

(UP031)